### DISCUSSION TITLE
Cant install backpack 6  on laravel 10

### DISCUSSION THREAD

**Discussion by k-wiese** [Laravel-Backpack/community-forum#777] [Q&A (Help)] [answered]
When i ran composer require backpack/crud i had these composer errors:
![image](https://github.com/Laravel-Backpack/community-forum/assets/75091948/ee478528-6d68-4a07-bc4e-9560747f4f59)

i used composer require backpack/crud -W and it worked but I cant install devtools
![image](https://github.com/Laravel-Backpack/community-forum/assets/75091948/c84f0202-8bed-47af-8e90-6693ffd7b354)

Tried few laravel versions and php 8.1, 8.2, 8.3

**Comment by pxpm** [comment, marked as answer]
Hey @k-wiese I am not aware of such package `carbonphp/carbon-doctrine-types`.

It seems one of our requirements is not compatible with the requirements of the mentioned package. It requires `doctrine/dbal` 4.0 while we install 3.0, AFAIK is also the version Laravel uses: https://github.com/laravel/framework/blob/484e9c2c78637e3484681d7569a0e3f878dcd594/composer.json#L98

As I understand, 4.0 still in development, at least from branch readme: https://github.com/doctrine/dbal/tree/4.0.x 

You can probably work around it by setting your `composer.json` to accept `minimum-stability: dev` and trying to force `doctrine/dbal: ^4.0`.

Alternatively use version `^2.0` of that package: https://github.com/CarbonPHP/carbon-doctrine-types/releases/tag/2.0.0 

We can only think in adding support for 4.0 when it's released as a stable version, in fact the last release `RC-1` had breaking changes, so it would be careless to add core support for a work in progress. https://github.com/doctrine/dbal/releases/tag/4.0.0-RC1

Let me know how it goes.

Cheers

**Comment by adriallongarriu** [comment]
The other day i was facing the same issue.
I ende up adding `"backpack/crud": "^6.4",` to composer manually and then  use `composer update`.
This donwngrade the version to 2.1
![image](https://github.com/Laravel-Backpack/community-forum/assets/20278289/d0170cfd-da13-4f44-be16-d43454e1b98e)

The key is `nesbot/carbon` is using  `"carbonphp/carbon-doctrine-types": "*",`  and whit the installation of laravel you install a version higher that backpack in my case installed the version 3.1
